### ISSUE TITLE
Unwanted line break when inserting curly braces

### ISSUE THREAD

**Issue by jmarca** [mooz/js2-mode#251]
First, I'm pretty sure this is an electric-layout-mode issue, but I only ever see it in js2-mode, so perhaps it is something being set here??  Second, I *use* emacs, but am pretty useless when it comes to programming elisp or really tweaking config files.

The issue:

In js2-mode, whenever I type a left curly brace `{`, emacs automatically inserts the closing brace and two linefeeds.  

For example, if I type:

```
var test = {
```

What I see is

```
var test = {
    
}
```

This is annoying, to say the least.  In preparing this issue, I figured out how to turn it off---by turning off electric-layout-mode.  However, according to "describe-mode",  my value of electric-layout-rules is nil, so I'm not doing anything to make this broken behavior.

With electric-layout off, when I type

```
var s = {
```

... I get ...

```
var s = {}
```

If I hit the right brace, no line break, and if I press enter rather than closing the braces, then emacs inserts two line breaks and I see:

```
var s = {

}
```

This is what I want.  Should I just globally turn off electric-layout-mode?

my emacs version is 24.5.1.

**Comment by dgutov** [comment]
> my value of electric-layout-rules is nil

In which buffer?

> Should I just globally turn off electric-layout-mode?

Why not? Do you use it in other modes? I don't.

**Comment by jmarca** [comment]
yep, setting the option for electric layout mode to off in the customization dialog for electricity killed it off.

I don't use it, but then, I don't *not* use it, right?  I just let the defaults fall where they may.  I only get this strange behavior in js2-mode, not in, say, ESS-mode, so I filed this just in case something was being set here.

Anyway, setting the default electric-layout-mode to "off" fixes this.

![emacsopts](https://cloud.githubusercontent.com/assets/70304/9070751/4d43cea6-3aa6-11e5-830f-202aaa000a50.jpg)

**Comment by dgutov** [comment]
> I just let the defaults fall where they may.

`electric-layout-mode` is disabled by default.

> not in, say, ESS-mode

Maybe because it doesn't provide the information required for `electric-layout-mode` to work (I have no idea if it does)?

**Comment by jmarca** [comment]
> electric-layout-mode is disabled by default.

Perhaps so, but for some reason electric-layout-mode was *not* disabled until I set it that way.  I didn't deliberately turn it on ever (I only discovered it today) but I did switch a while back from autopair to electric-pair-mode, so perhaps I flipped it on by accident then?  I don't know. Maybe if anybody else has this issue they will stumble across this and try turning off electric-layout-mode.
